src/PKSim.Core/Services/GlobalPKAnalysisTask.cs

```diff
@@ -30,136 +32,187 @@ public interface IGlobalPKAnalysisTask
       /// </summary>
       void CalculateDDIRatioFor(Simulation simulation, string compoundName);
+
+      IEnumerable<QuantityPKParameter> CalculateQuantityPKForPopulationSimulation(PopulationSimulation populationSimulation);
    }
 
    public class GlobalPKAnalysisTask : IGlobalPKAnalysisTask
    {
+      private readonly IPKAnalysesTask _pkAnalysesTask;
       private readonly IParameterFactory _parameterFactory;
       private readonly IProtocolToSchemaItemsMapper _protocolToSchemaItemsMapper;
       private readonly IProtocolFactory _protocolFactory;
       private readonly IGlobalPKAnalysisRunner _globalPKAnalysisRunner;
-      private readonly IPKAnalysesTask _pkAnalysisTask;
-      private readonly IPKCalculationOptionsFactory _pkCalculationOptionsFactory;
       private readonly IVSSCalculator _vssCalculator;
       private readonly IInteractionTask _interactionTask;
       private readonly ICloner _cloner;
+      private readonly IPKCalculationOptionsFactory _pkCalculationOptionsFactory;
+      private readonly IPopulationSimulationBodyWeightUpdater _populationSimulationBodyWeightUpdater;
 
-      public GlobalPKAnalysisTask(IParameterFactory parameterFactory, IProtocolToSchemaItemsMapper protocolToSchemaItemsMapper,
-         IProtocolFactory protocolFactory, IGlobalPKAnalysisRunner globalPKAnalysisRunner, IPKAnalysesTask pkAnalysesTask,
-         IPKCalculationOptionsFactory pkCalculationOptionsFactory, IVSSCalculator vssCalculator, IInteractionTask interactionTask, ICloner cloner)
+      public GlobalPKAnalysisTask(
+         IParameterFactory parameterFactory, IProtocolToSchemaItemsMapper protocolToSchemaItemsMapper, IProtocolFactory protocolFactory, 
+         IGlobalPKAnalysisRunner globalPKAnalysisRunner, IPKAnalysesTask pkAnalysesTask, IPKCalculationOptionsFactory pkCalculationOptionsFactory, 
+         IVSSCalculator vssCalculator, IInteractionTask interactionTask, ICloner cloner, IPopulationSimulationBodyWeightUpdater populationSimulationBodyWeightUpdater)
       {
+         _pkAnalysesTask = pkAnalysesTask;
          _parameterFactory = parameterFactory;
          _protocolToSchemaItemsMapper = protocolToSchemaItemsMapper;
          _protocolFactory = protocolFactory;
          _globalPKAnalysisRunner = globalPKAnalysisRunner;
-         _pkAnalysisTask = pkAnalysesTask;
-         _pkCalculationOptionsFactory = pkCalculationOptionsFactory;
+
          _vssCalculator = vssCalculator;
          _interactionTask = interactionTask;
          _cloner = cloner;
+         _pkCalculationOptionsFactory = pkCalculationOptionsFactory;
+         _populationSimulationBodyWeightUpdater = populationSimulationBodyWeightUpdater;
       }
 
-      private GlobalPKAnalysis calculateGlobalPKAnalysisFor(Simulation simulation)
+      private IContainer calculateGlobalPKAnalysisFor(Simulation simulation, Compound compound, DataColumn peripheralVenousBloodCurve, DataColumn venousBloodCurve, PKCalculationOptions options)
```

Comment:
This method is now called per compound so that it was easier to do the body weight scaling. The content should be the same otherwise.